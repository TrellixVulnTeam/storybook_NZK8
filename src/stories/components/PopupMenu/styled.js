// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

/**
 * Determine the button's accent color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getAccentColor(theme) {
  switch (theme) {
    case 'gray':
      return colors.paleGrey;
    case 'blue':
      return colors.brightBlue;
    case 'green':
      return colors.shamrockGreen;
    case 'yellow':
      return colors.burntYellow;
    default:
      return colors.strawberry;
  }
}

export const Wrapper = styled.span`
  color: ${colors.white};
  font-size: 13px;
  font-weight: ${p => (p.thin ? 400 : 600)};
  border-radius: 6px;
  padding: ${p => (p.thin ? '2px 4px' : '3px 5px')};
  text-align: center;
  border: none;
  background-color: ${p => getAccentColor(p.theme)};
  opacity: ${p => (!p.loading && p.disabled ? 0.5 : 1)};
  letter-spacing: -0.34px;
`;
