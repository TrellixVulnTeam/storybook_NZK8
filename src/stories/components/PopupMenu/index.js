// @flow

import React from 'react';
import { Wrapper } from './styled';

type Props = {
  children: React$Node,
  theme?: 'gray' | 'blue' | 'red' | 'green' | 'yellow',
  thin?: boolean,
};

export default (props: Props) => (
  <Wrapper theme={props.theme} thin={props.thin}>
    {props.children}
  </Wrapper>
);
