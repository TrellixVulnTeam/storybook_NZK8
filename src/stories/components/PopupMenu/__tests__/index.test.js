import React from 'react';
import { mount } from 'enzyme';
import PopupMenu from 'components/PopupMenu/index';

describe('<PopupMenu />', () => {
  it('should render correctly without props', () => {
    const sut = mount(<PopupMenu />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with blue theme', () => {
    const sut = mount(<PopupMenu theme="blue" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with red theme', () => {
    const sut = mount(<PopupMenu theme="red" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with green theme', () => {
    const sut = mount(<PopupMenu theme="green" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with yellow theme', () => {
    const sut = mount(<PopupMenu theme="yellow" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with gray theme', () => {
    const sut = mount(<PopupMenu theme="gray" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly if thin', () => {
    const sut = mount(<PopupMenu thin />);

    expect(sut).toMatchSnapshot();
  });
});
