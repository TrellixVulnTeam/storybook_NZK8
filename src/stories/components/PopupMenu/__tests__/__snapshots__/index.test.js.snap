// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupMenu /> should render correctly if thin 1`] = `
<_default
  thin={true}
>
  <styled__Wrapper
    thin={true}
  >
    <span
      className="styled__Wrapper-sc-1tz7lxd-0 eWawnb"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<PopupMenu /> should render correctly with blue theme 1`] = `
<_default
  theme="blue"
>
  <styled__Wrapper
    theme="blue"
  >
    <span
      className="styled__Wrapper-sc-1tz7lxd-0 kxfMAx"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<PopupMenu /> should render correctly with gray theme 1`] = `
<_default
  theme="gray"
>
  <styled__Wrapper
    theme="gray"
  >
    <span
      className="styled__Wrapper-sc-1tz7lxd-0 ehXENz"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<PopupMenu /> should render correctly with green theme 1`] = `
<_default
  theme="green"
>
  <styled__Wrapper
    theme="green"
  >
    <span
      className="styled__Wrapper-sc-1tz7lxd-0 iZebjb"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<PopupMenu /> should render correctly with red theme 1`] = `
<_default
  theme="red"
>
  <styled__Wrapper
    theme="red"
  >
    <span
      className="styled__Wrapper-sc-1tz7lxd-0 fcniRP"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<PopupMenu /> should render correctly with yellow theme 1`] = `
<_default
  theme="yellow"
>
  <styled__Wrapper
    theme="yellow"
  >
    <span
      className="styled__Wrapper-sc-1tz7lxd-0 lcvThl"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<PopupMenu /> should render correctly without props 1`] = `
<_default>
  <styled__Wrapper>
    <span
      className="styled__Wrapper-sc-1tz7lxd-0 fcniRP"
    />
  </styled__Wrapper>
</_default>
`;
